tool/gravity/cli/config.go

```diff
@@ -335,6 +335,11 @@ func (i *InstallConfig) CheckAndSetDefaults() (err error) {
 	if err := checkLocalAddr(i.AdvertiseAddr); err != nil {
 		return trace.Wrap(err)
 	}
+
+	if err := checkOverlayNet(i.AdvertiseAddr, i.ServiceCIDR, i.PodCIDR); err != nil {
```

Comment:
Nit: "overlay" usually refers only to the pod network, not service network, so I'd rename this method to something more generic, e.g. `checkNetworkConflict` or something.